edge-modules/iotedge-diagnostics-dotnet/src/Program.cs

```diff
@@ -91,10 +91,13 @@ static async Task Upstream(string hostname, string port, string proxy)
                 var httpRequest = new HttpRequestMessage(HttpMethod.Get, logsUrl);
                 HttpResponseMessage httpResponseMessage = await httpClient.SendAsync(httpRequest, HttpCompletionOption.ResponseHeadersRead);
 
-                var keys = httpResponseMessage.Headers.GetValues("iothub-errorcode");
-                if (!keys.Contains("InvalidProtocolVersion"))
+                if (isNested != null)
```

Comment:
>if (isNested != null) [](start = 16, length = 21)

for the non-nested case we prob still need to at least check that the response isn't a 404 or a timeout.

Reply:
That's the same logic as before. It trigger an exception for timeout I tested that. If it gets a 404, it still means it succeeded in connecting to iotHub since something replied. I am a bit wary of changing the logic so close it release since I can't test all the outliers.
In this PR I want to focus on fixing a specific bug.